### PR TITLE
[vividus-plugin-db] hash method is independent of key set order

### LINKED ISSUE #769: DB polling step constructs result table with columns not in selected order
**Steps to reproduce**
1. Compose query with non alphabetical order of columns required:
```
When I initialize the SCENARIO variable `sqlQuery` with value `
  select store_id, sku_id from scheme.table_name
  where sku_id = 1234
    and store_id = 4321
    and rownum = 1`
```
2a. Query DB and expect table with columns not in alphabetical order:
```
When I wait for 'PT10S' duration retrying 1 times while data from `${sqlQuery}` executed against `dbName` is equal to data from:
|STORE_ID |SKU_ID  |
|4321     |1234    |
```
2b. Query DB and expect table with columns in alphabetical order:
```
When I wait for 'PT10S' duration retrying 1 times while data from `${sqlQuery}` executed against `dbName` is equal to data from:
|SKU_ID  |STORE_ID |
|1234    |4321     |
```
**Expected:** 2a passed, 2b failed OR both passed (because column names were provided)
**Actual:** 2a failed, 2b passed

### PR DESCRIPTION
**Pull request by VitaliLebedzeu**
Fixes #769

### CODE REVIEW

File: vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java [hunk 1]
```diff
@@ -418,6 +418,7 @@ private HashCode hash(Set<String> keys, Map<String, Object> map)
                            .stream()
                            .map(map::get)
                            .filter(Objects::nonNull)
+                           .sorted()
```
Comment (valfirst) [inline vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:421]: what is root cause of the issue?

File: vividus-plugin-db/src/test/java/org/vividus/bdd/steps/db/DatabaseStepsTests.java [hunk 2]
```diff
@@ -373,13 +373,21 @@ void shouldThrowTimeoutExceptionIfQueryTakesTooMuchTime()
         verifyNoInteractions(attachmentPublisher, softAssert);
     }
 
-    @Test
-    void shouldCompareDataVsExamplesTableAndNotPostReportIfDataEqual()
+    static Stream<Arguments> equalDataProvider() {
```
Comment (valfirst) [inline vividus-plugin-db/src/test/java/org/vividus/bdd/steps/db/DatabaseStepsTests.java:376]: please fix checkstyle violations

Comment (VitaliLebedzeu) [reply vividus-plugin-db/src/test/java/org/vividus/bdd/steps/db/DatabaseStepsTests.java]: fixed

File: vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java [hunk 3]
```diff
@@ -249,7 +249,8 @@ public void compareData(String sourceSqlQuery, String sourceDbKey, String target
     }
 
     /**
-     * Step waits until data received from SQL request is equal to data from examples table for some duration.
+     * Step waits until data received from SQL request is equal to data from examples table in any order
```
Comment (ikalinin1) [inline vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:252]: Could you please clarify what's unordered? Columns within a row and rows within a result set?

Comment (ikalinin1) [reply vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:252]: and please add System test

Comment (VitaliLebedzeu) [reply vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java]: definition was changed
system test was added

File: vividus-tests/src/main/resources/story/system/DatabaseStepsTests.story [hunk 4]
```diff
@@ -24,3 +24,13 @@ Then `${result}` matching rows using `id` from `vividus` is equal to data from:
 |1 |Valery                   |
 |2 |Vlad                     |
 |3 |Ivan                     |
+
+Scenario: Verify step: 'I wait for '$duration' duration retrying $retryTimes times while data from `$sqlQuery` executed against `$dbKey` is equal to data from:$table"'
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/system/DatabaseStepsTests.story:28]: `When` is missing?

Comment (VitaliLebedzeu) [reply vividus-tests/src/main/resources/story/system/DatabaseStepsTests.story]: oops... you are right thanks.
fixed

File: vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java [hunk 5]
```diff
@@ -249,7 +249,8 @@ public void compareData(String sourceSqlQuery, String sourceDbKey, String target
     }
 
     /**
-     * Step waits until data received from SQL request is equal to data from examples table for some duration.
+     * Step waits until data received from SQL request is equal to data from examples table with any order columns
+     * within a row for some duration (each received data element CONTAINS in examples table).
```
[the remark this answers is not in the document]
Comment (ivan-lednev) [reply vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:253]:
``` 
   /**
     * The step waits until the <code>sqlQuery</code> returns the data equal to the examples table row
     * The order of columns is ignored
     * Actions:
     * <ul>
     *     <li>run the <code>sqlQuery</code></li>
     *     <li>compare the response with the examples table row</li>
     *     <li>sleep for the <code>duration</code></li>
     *     <li>repeat all of the above until there are no more retry attempts or the response is equal to the examples table row</li>
     * </ul>
     * @param duration The time gap between two queries
     * @param retryTimes How many times request will be retried; duration/retryTimes=timeout between requests
     * @param sqlQuery SQL query to execute
     * @param dbKey Key identifying the database connection
     * @param table Rows to compare the data against
```

[the remark this answers is not in the document]
Comment (VitaliLebedzeu) [reply vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java]: rewrite

**Comment by valfirst** [comment]
Please fix checkstyle violations

**Pull request merged**